api/v1/storagecluster_types.go

```diff
@@ -112,10 +112,12 @@ type StorageClusterSpec struct {
 }
 
 type SharedFilesystemConfigurationSpec struct {
+	// +kubebuilder:validation:Optional
 	Parameters map[string]string `json:"parameters,omitempty"`
 }
 
 type BlockPoolConfigurationSpec struct {
+	// +kubebuilder:validation:Optional
```

Comment:
I think this is not required because BlockPoolConfigurationSpec is already marked as optional. Same for filesystemconfigspec as well.

Reply:
Probably. Just wanted to be on the safe side of things